src/Router.js

```diff
@@ -58,14 +58,25 @@ export default class TipsiRouter {
 
   config() {}
 
-  push(e, route, paramsOrOptions = {}) {
+  callHistoryMethodWithArguments(methodName, e, locationHolder, paramsOrOptions = {}) {
     if (e) {
       e.preventDefault()
     }
-    const toPath = compile(route.path)
-    const path = toPath(paramsOrOptions)
-    const { config, ...params } = paramsOrOptions
-    this.history.push(path, params)
+    const { path, pathname } = locationHolder
+    if (path) {
+      const { config, ...params } = paramsOrOptions
+      this.history[methodName](compile(path)(paramsOrOptions), params)
+    } else if (pathname) {
+      this.history[methodName](locationHolder)
+    } else {
+      throw new Error(
+        `locationHolder arg for ${methodName}(...) has neither "path" nor "pathname" key`
+      )
+    }
+  }
+
+  push(e, locationHolder, paramsOrOptions = {}) {
```

Comment:
`route` was a good name for parameter, why `locationHolder` instead?

Reply:
"route" was too blurry definition for me. It has too many semantic clashes in tipsi-router, react-router, history so i just always asking myself "what kind of route is this route" =). I'll bring old name if you like.

Reply:
I just want to remember you — if you've changed a name for a thing in `tipsi-router` you should change it on every internal router